include/dt-bindings/gpio/gpio.h

```diff
@@ -102,6 +102,59 @@
 
 /** @} */
 
+/**
+ * @name GPIO drive strength flags
+ * The `GPIO_DS_*` flags are used with `gpio_pin_configure` to specify the drive
+ * strength configuration of a GPIO pin.
+ *
+ * The drive strength of individual pins can be configured
+ * independently for when the pin output is low and high.
+ *
+ * The `GPIO_DS_*_LOW` enumerations define the drive strength of a pin
+ * when output is low.
+
+ * The `GPIO_DS_*_HIGH` enumerations define the drive strength of a pin
+ * when output is high.
+ *
+ * The interface supports two different drive strengths:
+ * `DFLT` - The lowest drive strength supported by the HW
+ * `ALT` - The highest drive strength supported by the HW
+ *
+ * On hardware that supports only one standard drive strength, both
+ * `DFLT` and `ALT` have the same behavior.
+ * @{
+ */
+/** @cond INTERNAL_HIDDEN */
+#define GPIO_DS_LOW_POS 20
```

Comment:
gpio_dt_flags_t is currently defined as uint8_t.  gpio_dt_flags_t should be updated as part of this change.  

If the GPIO flags in devicetree are going to be extended past 8 bits, It probably makes sense to just eliminate gpio_dt_flags_t and only use gpio_flags_t throughout Zephyr.

Reply:
I have opted to extend `gpio_dt_flags_t` to 16 bits, but kept it for now. I agree that we should consider removing it, but I would like to keep that out of this PR.

Reply:
We (Google) have a use case for specifying the GPIO_INPUT/OUTPUT and the GPIO_OUTPUT_INIT_LOW/INIT_HIGH flags in the devicetree. So I'm in favor of extended gpio_dt_flags_t to 32 bits and making all the flags available in the bindings header.

Reply:
That would make all of our GPIO flags Zephyr-specific devicetree extensions. I am not sure, that's a good plan.

Reply:
I will bring this up on todays dev-review meeting.

Reply:
We discussed this a bit on todays API meeting, but didn't reach a conclusion.
@mbolivar-nordic Any input on this?

Reply:
> @mbolivar-nordic Any input on this?

I'd like to get @mnkp to approve this. It's always been his design decision as GPIO maintainer that the Zephyr devicetree GPIO flags match Linux's exactly, so I think this is his decision.

Reply:
@henrikbrixandersen could you please comment on the fact that we'd diverge from the Linux GPIO flags in DT?

Reply:
@carlescufi That's already present: `The drive strength flags are a Zephyr specific extension ...`